lib/iris/cube.py

```diff
@@ -1681,10 +1703,18 @@ def data(self):
             (10, 20)
 
         """
-        data = self._my_data
-        if is_lazy_data(data):
+        if self._numpy_array is None:
             try:
-                data = as_concrete_data(data)
+                data = self._dask_array.compute()
```

Comment:
See this [comment](https://github.com/SciTools/iris/pull/2392/files#r102430258) regarding use of `num_workers`